app/src/viewmodels/camera/CameraViewModel.tsx

```diff
@@ -90,13 +79,47 @@ export default function useCameraViewModel(
     }
   };
 
-  // Generate an image URL from the picture taken
-  const imageUrlGen = async () => {
-    const img_id = await firestoreCtrl.uploadImageFromUri(picture?.uri);
-    navigation.navigate("CreateChallenge", {
-      image_id: img_id,
-      group_id: group_id,
-    });
+  // Toggle location switch
+  const toggleLocation = () => setIsLocationEnabled((prev) => !prev);
+
+  // Fetch the current location
+  useEffect(() => {
+    async function fetchLocation() {
+      let { status } = await requestForegroundPermissionsAsync();
+      if (status !== "granted") {
+        setIsLocationEnabled(false);
+        return;
+      }
+      let currentLocation = await getCurrentPositionAsync();
+      setLocation(currentLocation);
+    }
+    fetchLocation();
+  }, []);
+
+  // Create the challenge
+  const makeChallenge = async () => {
+    try {
+      const imageId = await firestoreCtrl.uploadImageFromUri(picture?.uri);
+      const date = new Date();
+      await createChallenge(
+        firestoreCtrl,
+        group_id,
+        description,
+        isLocationEnabled ? location : null,
+        group_id,
+        date,
+        imageId,
+      );
+      if (group_id == "" || group_id == "home") {
```

Comment:
Why is there the option group_id == ""?

Reply:
It was already there, I think it's legacy from some posts that can have a no group_id.

Reply:
Alright !